### ISSUE TITLE
Variables: option to only return enabled Zabbix hosts

### ISSUE THREAD

**Issue by Rickkwa** [grafana/grafana-zabbix#362]
#### I'm submitting a ...
- [ ] Bug report
- [x] Feature request
- [ ] Question / Support request

Hello,
Right now when you create a template variable and your query, for example, is simply `*.*` then it will return all hosts. Even the hosts that are in `disabled` state are returned. I think an option to return only `enabled` hosts would be nice to have.

Thanks.

**Comment by alexanderzobnin** [comment]
Sorry, but I can't add this option because datasource hasn't control over template variable editor. May be I can add it as data source option, but it will affect all variables belonging to given datasource.

**Comment by Rickkwa** [comment]
I think even if it affects all variables in the datasource, it is a nice option to have. Thanks.

**Comment by Perlovka** [comment]
Maybe it could be implemented in metrics, like "Show disabled items" one?

**Comment by alexanderzobnin** [comment]
But why do you need to hide disabled hosts? Because by default only enabled Items will be shown.

**Comment by ptempier** [comment]
Hi

I think it still shows if at the template/host level the item is disabled , but the  host itself is active

I get server versions as a table in grafana, but for disabled servers, i would get an empty line in grafana.

I also get an empty line for server with an unsupported items or one where the scripts bug parsing the version and return empty data. In that case i need to fix my script or whatever.

But if i disabled  the host in zabbix, this means i don't want/care about that data anymore and just want to keep the archived data for future reference.

To me it would seems fair to allow for a filter at the datasource level , if this can not be done as a grafana metric option.(at worst we could create a datasource for all hosts and one for active only hosts)

**Comment by urusha** [comment]
> But why do you need to hide disabled hosts? Because by default only enabled Items will be shown.

Because if there are many disabled hosts, filtering by hosts is really inconvenient, since you have to scroll down again and again.

**Comment by alexanderzobnin** [comment]
@urusha you could just start to type hostname and reduce the list of suggested hosts.

**Comment by urusha** [comment]
@alexanderzobnin I could, but is it convenient? If I see 5 hosts with different names [a-z] instead of 100, I could easily mark them in a few secs without any typing... More over, if I get no data after choosing the particular host, I can not even determine if the host is enabled or not, or if the item is enabled or not (disabled / unsupported / has no values).
That would be great if we could use `filter` parameter in the api queries somehow (`status` in this case).

**Comment by polak785** [comment]
Indeed, like other ones, when we have many servers and the old ones are only deactivated and not deleted on Zabbix, this is very annoying to scroll down on Grafana on dashbords that display metrics with repetitive panels based on hosts.

Will it be supported in the future?

**Comment by skaboy007** [comment]
ye this one is actually extremely important

**Comment by JoyHusky** [comment]
I am using "Repeat options" -> "Repeat by variable" for auto create panels in dashbord.
And I fully agree with TS about necessity of this feature.

**Comment by polak785** [comment]
Please re-open it. This is a feature request in the top requested (regarding the number of 👍)
